test/integration.js

```diff
@@ -16,14 +19,20 @@ test('integration', async () => {
 
 test('retrieval check for our CID', async () => {
   const spark = new Spark()
-  spark.getRetrieval = async () => ({ cid: KNOWN_CID })
+  spark.getRetrieval = async () => ({ cid: KNOWN_CID, minerId: OUR_FAKE_MINER_ID })
+  spark.lookupMinerPeerId = async (minerId) => {
```

Comment:
In this particular test, if my stubbed function isn't called, the retrieval check fails because Filecoin's RPC API call will reject the fake miner id.

The purpose of this test is not to verify how Spark works under the hood (and whether it calls all other functions we think it should call) but whether it can find the provider and retrieve the data for the CID served by our Frisbii instance.

However, if you think it's worth asserting that both functions were called, then I am fine with adding that. Let me know!

Reply:
This was also my understanding when I wrote tests like this, but through your reviews I understood that it's clearer and safer to assert the function calls. If in the future a condition is introduced which skips `lookupMinerPeerId()`, then this test will still pass.

However,iIn my previous review I missed that this was an integration test and not a unit test. Please choose the testing method that you see fits best.

Reply:
> push the function calls into an array, and then assert this function was called

added in 0a48986